src/stan/math/prim/scal/meta/length_mvt.hpp

```diff
@@ -9,16 +9,12 @@ namespace stan {
 
   // length_mvt() should only be applied to std vector or Eigen matrix
   template <typename T>
-  size_t length_mvt(const T& ) {
-    throw std::out_of_range("length_mvt passed to an unrecognized type.");
```

Comment:
Agreed, but the problem is that we'll need a generic template function to
pick all the possible instantiations. I'm guessing this works because we're
missing some tests.

On Wed, Mar 11, 2015 at 8:20 PM, Bob Carpenter <notifications@github.com>
wrote:

> In src/stan/math/prim/scal/meta/length_mvt.hpp
> <https://github.com/stan-dev/stan/pull/1210#discussion_r26269160>:
>
> > @@ -9,16 +9,12 @@ namespace stan {
> >
> >    // length_mvt() should only be applied to std vector or Eigen matrix
> >    template <typename T>
> > -  size_t length_mvt(const T& ) {
> > -    throw std::out_of_range("length_mvt passed to an unrecognized type.");
>
> Absolutely better to check at compile-time if possible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stan-dev/stan/pull/1210/files#r26269160>.
>